js/Worksheet.js

```diff
@@ -1087,6 +1135,120 @@ class Worksheet extends HTMLElement {
         return CSVParser.unparse(data);
     }
 
+    fromExcel(aString, rABS, fileName, ftype){
```

Comment:
One issue I've encountered here is that -- unlike `fromCSV` -- this function does not update the size of the dataFrame of the current Worksheet. If the incoming data is bigger than the current dataFrame an error is thrown and nothing is loaded:

![Screen Shot 2022-09-22 at 7 18 55 AM](https://user-images.githubusercontent.com/1640299/191733473-1eef55c1-1a67-4ed4-a4b5-d9a76ad84b8b.png)

Reply:
@darth-cheney right, I forgot about this. There was a related issue with fromCSV. There is no guarantee in exports from MS/OpenOffice that the underlying nd-array (for .csv/.xlsx) is uniform, ie has the same length arrays/rows. 

From what I can see this is almost never the case, and what you see in sheet-js is an nd-array where each row-array is the minimum length necessary for the corresponding row, like the column-array length is the min for the colum, ie a maximally conservative approach. 

So if you were to upload a csv or excel with row data lengths [2, 10] then current dataFrame.corner.x = 2 and an error was thrown. 

There is also another [issue](https://github.com/darth-cheney/ap-sheet/issues/31). If you set the dataFrame size smaller than the size of the frame in view everything breaks. Similarly if you try to resize a sheet outside of the dataFrame size everything breaks. 

The [temp solution just pushed](https://github.com/dkrasner/worksheet/pull/39/commits/b27e2a3d9d35ffa7b4570552570951b0648ff008) is not going to solve the problem. But I think this should be solved at the sheet level, ie the sheet should be able to take whatever data is given it and grow when asked to etc. Worksheet doesn't need to know how sheet handles data, it's just a wrapper.